### PR TITLE
Rov 45

### PR DESCRIPTION
**Pull request by DheerajGBhatt**
# Description

Separated the generateSAM module. 

## Issue ticket number and link
[ROV-45](https://github.com/antstackio/rover-engine/pull/48)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## New Packages

1.

## Packages Removed 

1. esm
2. ini
3. js-yaml(JS)

### CODE REVIEW

File: generateSAM/generateSAM.ts [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+        data["AWSTemplateFormatVersion"] =
+          config.SkeletonConfig["template_version"];
+      if (!Object.prototype.hasOwnProperty.call(data, "Resources"))
+        throw new Error("Improper SAM template file in " + filename);
+    }
+  
+    const template = addResourceTemplate(stackes, stack_names, data);
+    const doc = new yaml.Document();
+    doc.contents = template;
+    utlities.writeFile(app_data.app_name + "/template.yaml", doc.toString());
```
Comment (hnprashanth) [inline generateSAM/generateSAM.ts]: It might not be a synchronous call - check implications

File: generateSAM/generateSAM.ts [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+        ](ele, input["stack_details"][ele]["params"]);
+      } else if (input["stack_details"][ele]["type"] == "Custom") {
+        const resources: AnyArray = [];
+        const customstackarray: AnyArray =
+          input["stack_details"][ele]["componentlist"];
+        customstackarray.map((ele) => {
+          const componentarray: AnyArray = JSON.parse(
+            JSON.stringify(components.Components[ele])
+          );
+          componentarray.map((ele) => {
```
Comment (hnprashanth) [inline generateSAM/generateSAM.ts:62]: variable naming convention

File: generateSAM/generateSAM.ts [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+        app_types[ele]["resources"] = resources;
+        app_types[ele]["type"] = "components";
+      } else {
+        stackdata = JSON.parse(
+          JSON.stringify(
+            modules.Modules[input["stack_details"][ele]["type"]]["resource"]
+          )
+        );
+      }
+      Object.keys(stackdata).forEach((ele1) => {
```
Comment (hnprashanth) [inline generateSAM/generateSAM.ts:76]: variable naming convention

File: utlities/utilities.ts [hunk 4]
```diff
@@ -5,21 +5,20 @@ import * as modules from "../resources/modules";
...
+
+const exec = child.execSync;
 const sub = new RegExp(
   /(!Sub|!Transform|!Split|!Join|!Select|!FindInMap|!GetAtt|!GetAZs|!ImportValue|!Ref)\s[a-zA-Z0-9 !@#$%^&*()_+\-=[\]{};':"\\|,.<>/?]*\n/g
 );
 const pythonpattern = new RegExp(/python[1-9]*\.[1-9]*/g);
 const jspattern = new RegExp(/nodejs[1-9]*\.[a-zA-Z]*/g);
 const yamlpattern = new RegExp(/(\.yaml$)/g);
-const Yaml = require("js-yaml");
+import * as Yaml from "js-yaml";
```
Comment (KrishnaMuddi) [inline utlities/utilities.ts:19]: js-yaml package seems to be removed from package json file, please check

**Pull request merged**